share/spice/airlines/airlines.js

```diff
@@ -1,115 +1,255 @@
+        }
+
+        // Add the date and time or departure or arrival.
+        Spice.registerHelper("airline_time", function(isDeparture, arrivalDate, departureDate) {
+            var dateObject = new Date(arrivalDate);
+            if(isDeparture) {
+                dateObject = new Date(departureDate);
+            }
+
+            var hours = dateObject.getHours(),
+                minutes = dateObject.getMinutes(),
+                date = dateObject.toDateString();
+
+            date = date.substring(0, date.lastIndexOf(" "));
+
+            // AM or PM?
+            var suffix = (hours >= 12) ? "PM" : "AM";
+
+            // Convert to 12-hour time.
+            if(hours > 12) {
+                hours -= 12;
+            } else if(hours === 0) {
+                hours = 12;
+            }
+
+            // Add leading zeroes.
+            minutes = minutes < 10 ? "0" + minutes : minutes;
+
+            if ((arrivalDate.getDate() != departureDate.getDate()) && (!isDeparture)) {
+                return hours + ":" + minutes + " " + suffix + " (" + MONTH[dateObject.getMonth()] + " " + dateObject.getDate() + ")";
+            } else
+                return hours + ":" + minutes + " " + suffix;
+        });
+
+        Spice.registerHelper("airline_status", function(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival) {
+            var result = onTime(flight, departureDate, arrivalDate, scheduledDeparture, scheduledArrival),
+                ok_class = result[1] ? "tile__ok" : "tile__not";
+            return '<div class="' + ok_class + '">' + result[0] + '</div>';
+        });
 
         Spice.add({
             data: results,
             id: "airlines",
-            name: "Flights",
+            name: "Airlines",
             meta: {
                 minItemsForModeSwitch: 3,
                 sourceName: 'FlightStats',
-                sourceUrl: source,
-                itemType: DDG.capitalizeWords(flight[0].Airline.Name) + " Flights"
+                sourceUrl: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?"
+                    + "airlineCode=" + results[0].airlineCode 
+                    + "&flightNumber=" + results[0].flightNumber,
+                itemType: results.length === 1 ? "flight" : "flights by departure time" 
             },
             normalize: function(item) {
+
                 return {
-                    url: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?&airlineCode=" +
-                        item.flight.Airline.AirlineCode +
-                        "&flightNumber=" +
-                        item.flight.FlightNumber
+                    url: "http://www.flightstats.com/go/FlightStatus/flightStatusByFlight.do?"
+                        + "airlineCode=" + item.airlineCode 
+                        + "&flightNumber=" + item.flightNumber
                 }
             },
             sort_fields: {
```

Comment:
I was wrong--there could be multiple flights and it's good to sort them by date. 😬